client/src/main/java/com/paypal/selion/internal/platform/grid/browsercapabilities/AdditionalSauceCapabilitiesBuilder.java

```diff
@@ -49,12 +49,13 @@ public DesiredCapabilities getCapabilities(DesiredCapabilities capabilities) {
             return capabilities;
         }
 
-        capabilities = appendSauceLabsCredentials(capabilities);
-        capabilities = appendSeleniumVersion(capabilities);
-        capabilities = appendSauceLabsCapabilities(capabilities);
+        DesiredCapabilities dc = new DesiredCapabilities();
```

Comment:
This doesn't really solve the problem I suspect codacy was reporting which probably had something to do with `capabilities` being a parameter and getting re-assigned. Each call to `appendSauceLabsCredentials`, etc here here will return a new set of `capabiltiies` which `dc` will receive .. We need the result of the calls to merge/append the previous capabilities with the new ones.

How about

```java
DesiredCapabilities dc = capabilities.merge(appendSauceLabsCapabilities(capabilities))
    .merge(appendSeleniumVersion(capabilities))
    .merge(appendSauceLabsCapabilities(capabilities));
//...
return dc;
```